### PR TITLE
Separate Uploads and Datasets in two distinct concepts

### PR DESCRIPTION
**Pull request by yllieth**
... and also:
- remove dependency to errorManager service defined in ML|Studio implementation
- use angular's builtin events instead of eventEmitter lib (while this SDK is tightly coupled with angular)
- prepare the introduction of a new storage type by the API
- refactor the pseudo-resource Uploads within a helper and existing resources (Sources+Datasets)

https://github.com/PredicSis/hub/pull/549

### CODE REVIEW

File: dist/predicsis-jsSDK.js [hunk 1]
```diff
@@ -318,6 +317,19 @@ angular
         });
     };
 
+    this.createFromUpload = function(finishedUpload) {
```
Comment (dehau) [inline dist/predicsis-jsSDK.js]: Can you comment the interface of this function ?

File: lib/helper/uploadHelper.js [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+        if(xhr2.status === http.CREATED) {
+          $rootScope.$broadcast('jsSDK.upload.uploaded', uploadObject);
+        } else {
+          $rootScope.$broadcast('jsSDK.upload.error', { upload: uploadObject, request: xhr2 });
+        }
+      });
+
+      xhr2.addEventListener('error', function() {
+        uploadObject.isUploading = false;
```
Comment (dehau) [inline lib/helper/uploadHelper.js:57]: Can you add a `delete concurrentUploads[uploadObject.id]` ?

File: dist/predicsis-jsSDK.js (hunk 1 above)
Comment (yllieth) [inline dist/predicsis-jsSDK.js]: sure !

File: lib/helper/uploadHelper.js (hunk 2 above)
Comment (yllieth) [inline lib/helper/uploadHelper.js:57]: sure !

**Pull request merged**